lib/replicator.js

```diff
@@ -1157,7 +1157,10 @@ class Peer {
     }
 
     if (start === 0 && drop === false) {
-      if (length > this._remoteContiguousLength) this._remoteContiguousLength = length
+      if (length > this._remoteContiguousLength) {
+        this._remoteContiguousLength = length
+        this.core.updateRemoteContiguousLength(length)
```

Comment:
I'm not sure we can let it go uncaught either.

I think either we need a dedicated error handler, but we don't have any standard place to emit the event.. 

What do you think about changing `updateRemoteContiguousLength` to be sync and have it set a `needsHintFlush` flag on the core, then `SessionState.flush` always checks that flag and when flushed [these lines](https://github.com/holepunchto/hypercore/pull/743/files#diff-826e5ed9069c92baa50ec46dde8074a6789b4c90837c5ede0e6b0438e3c34309R401-R403) are triggered by calling a `_onhintflush` hook instead?

Reply:
Fixed in https://github.com/holepunchto/hypercore/pull/743/commits/52a3e890385ce516faf710704e69d447b9a262c4